### PR TITLE
?status added that returns proper HTTP code on ok, failure or error

### PR DESCRIPTION
**Pull request by leCandas**
- New request param "status" introduced for general check by http status
- New format "http" intorduced for simple status check.

### CODE REVIEW

File: src/main/java/com/philemonworks/selfdiagnose/output/HTTPReporter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import java.io.InputStreamReader;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Iterator;
+
+/**
+ * HTMLReporter is to produce an HTTP report for a DiagnoseRun.
+ *
+ */
+public class HTTPReporter implements DiagnoseRunReporter {
```
Comment (emicklei) [inline src/main/java/com/philemonworks/selfdiagnose/output/HTTPReporter.java]: Better call this StatusReporter as it does not actually report with a HTTP response.

File: src/main/java/com/philemonworks/selfdiagnose/SelfDiagnoseServlet.java [hunk 2]
```diff
@@ -151,6 +174,8 @@ public DiagnoseRunReporter getReporter(String format) {
             return new XMLReporter();
         if ("json".equals(format))
             return new JSONReporter();
+        if ("http".equals(format))
```
Comment (emicklei) [inline src/main/java/com/philemonworks/selfdiagnose/SelfDiagnoseServlet.java]: why did you choose to have 2 query parameters, status and http?

Comment (leCandas) [reply src/main/java/com/philemonworks/selfdiagnose/SelfDiagnoseServlet.java]: removed.

File: src/main/java/com/philemonworks/selfdiagnose/output/HTTPReporter.java (hunk 1 above)
Comment (leCandas) [reply src/main/java/com/philemonworks/selfdiagnose/output/HTTPReporter.java]: Changed.

**Pull request merged**